File: backend/app/Models/Business.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Business extends Model
{
    protected $fillable = [
        'name', 'address', 'phone', 'type', 'longitude', 'latitude', 'user_id'
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function products(): HasMany
    {
        return $this->hasMany(Product::class);
    }

    public function employeeRoles(): HasMany
    {
        return $this->hasMany(EmployeeRole::class);
    }
}

File: kkn_backend/app/models/business.go
package models

import (
	"github.com/goravel/framework/database/orm"
)

type BusinessType string

const (
	BusinessTypeRestoran   BusinessType = "restoran"
	BusinessTypeToko       BusinessType = "toko"
	BusinessTypeKantor     BusinessType = "kantor"
	BusinessTypeGudang     BusinessType = "gudang"
	BusinessTypePariwisata BusinessType = "pariwisata"
	BusinessTypeLainLain   BusinessType = "lain-lain"
)

type Business struct {
	orm.Model

	Longitude float64
	Latitude  float64
	Name      string
	Address   string
	Phone     string
	Type      BusinessType
	UserId    uint64

	User          User
	Products      []Product
	EmployeeRoles []EmployeeRole

	Images      []Image      `gorm:"polymorphic:Imageable;"`
}
